### PR TITLE
Change space and wrapping exclusions to include information about ignoring elastic

### PR DESCRIPTION
**Pull request by ryzngard**
Fixes: #30518, which was caused by elastic spacing being ignored in an array initializer during formatting. 

We were previously detecting elastic inclusion by verifying the first and last node with `ContainsElasticTrivia`. Instead, remove that function and update `IsSpacingSuppressed` and `IsWrappingSuppressed` to take a bool indicating that elastic spacing/wrapping should be ignored. These methods correctly identify if elastic trivia exists. 

Add a test to verify this doesn't regress in the future

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTests.cs [hunk 1]
```diff
@@ -1650,6 +1652,33 @@ public async Task FormatArbitraryNodeParenthesizedLambdaExpression()
...
+Identifier Identifier,
+    1
+}";
+
+            var tree = SyntaxFactory.ParseSyntaxTree(code, options: TestOptions.Script);
+            var root = tree.GetRoot();
+
+            var declaration = SyntaxFactory.DeclarationExpression(SyntaxFactory.IdentifierName(SyntaxFactory.Identifier("Identifier")), SyntaxFactory.SingleVariableDesignation(SyntaxFactory.Identifier("Identifier")));
+
+            var newType = declaration.Type.WithoutTrivia().WithTrailingTrivia(SyntaxFactory.ElasticMarker).WithAdditionalAnnotations(Formatter.Annotation);
```
Comment (heejaechang) [inline src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTests.cs]: you probably don't need this?

File: src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTests.cs [hunk 2]
```diff
@@ -1650,6 +1652,33 @@ public async Task FormatArbitraryNodeParenthesizedLambdaExpression()
...
+            var strippedDeclaration = SyntaxFactory.DeclarationExpression(newType, declaration.Designation);
```
Comment (heejaechang) [inline src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTests.cs]: declearation is already created from factory, so code below doesn't look like needed.

**Comment by heejaechang** [comment]
let's see why some tests are failed.

**Comment by ryzngard** [comment]
@heejaechang ping :)

**Comment by jinujoseph** [comment]
@ryzngard , @heejaechang  is this ready to merge ?

**Comment by ryzngard** [comment]
I'm double checking with late code and I'll merge tonight

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Pull request merged**